src/main/java/com/scottlogic/filters/AuthorPostFilter.java

```diff
@@ -0,0 +1,28 @@
+package com.scottlogic.filters;
+
+import com.scottlogic.PostFilter;
+import com.scottlogic.UserPost;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class AuthorPostFilter implements PostFilter {
+
+    final String Author;
+
+    public AuthorPostFilter(String authorInput) {
+        Author = authorInput;
+    }
+
+    @Override
+    public List<UserPost> filter(List<UserPost> inputList) {
+        if (inputList == null) {
+            return null;
+        }
+        if(Objects.equals(Author, "")){return new ArrayList<>();}
```

Comment:
This is a reasonable way of handling the empty string, the only slight issue with it is that it is inconsistent and not immediately obvious whether this is going to return all of the list or just an empty list. This means that in some designs you wouldn't be able to make a filter like this and an error would be thrown in the constructor instead. Both approaches have their pros and cons and this approach is perfectly reasonable just thought I should mention the other ways that Java sometimes does things

Reply:
Yeah that's useful, thanks